hop/configure.py

```diff
@@ -55,22 +61,82 @@ def _add_parser_args(parser):
         "-i", "--import", dest="import_cred", help="Import credentials from CSV file",
     )
 
+    subparser.add_parser("list-creds", help="Display all stored credentials")
+
+    add_cred_parser = subparser.add_parser("add-cred", help="Store an additional credential, "
+                                           "specified either via a CSV file or interactively")
+    add_cred_parser.add_argument("cred_file", type=str, default=None, nargs='?',
+                                 help="Import credentials from CSV file")
+
+    del_cred_parser = subparser.add_parser("delete-cred", help="Delete a stored credential")
+    del_cred_parser.add_argument("name", type=str,
+                                 help="The name of the credential to delete. This will be treated "
+                                 "as a username, unless it contains a @ character, in which case it"
+                                 " will be split into a username and hostname.")
+
+
+def read_new_credential(csv_file=None):
+    """Import a credential from a CSV file or obtain it interactively from the user.
+
+    Args:
+        csv_file: Path to a file from which to read credential data in CSV format.
+                  If unspecified, the user will be prompted to enter data instead.
+
+    Returns:
+        A configured `auth.Auth` object containing the new credential.
+
+    Raises:
+        FileNotFoundError: If csv_file is not None and refers to a nonexistent path.
+        KeyError: If csv_file is not None and the specified file does not contain either a username
+                  or password field.
+        RuntimeError: If csv_file is None and the interactively entered username or passwod is
+                      empty.
+
+    """
+    if csv_file is None:
+        logger.info("Generating configuration with user-specified username + password")
+        username = input("Username: ")
+        if len(username) == 0:
+            raise RuntimeError("Username may not be empty")
+        password = getpass.getpass()
+        if len(password) == 0:
+            raise RuntimeError("Password may not be empty")
+        hostname = input("Hostname (may be empty): ")
+    else:
+        if os.path.exists(csv_file):
+            with open(csv_file, "r") as f:
+                reader = csv.DictReader(f)
+                cred = next(reader)
+                username = cred["username"]
+                password = cred["password"]
+                hostname = cred["hostname"] if "hostname" in cred else ""
+        else:
+            raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), csv_file)
+    return auth.Auth(username, password, hostname)
+
+
+def write_auth_data(config_file, credentials):
+    """Write configuration file for the set of credentials.
 
-def write_config_file(config_file, username, password):
-    """Write configuration file for the given username and password.
+    Creates containing directories as needed.
```

Comment:
If any of the parent directories of `config_file` does not exist, it will be created. This comes up when we first create `~/.config/hop/auth.toml` and `~/.config/hop` does not yet exist.

Reply:
Ah gotcha, that makes sense, thanks.